waiter/src/waiter/service_description.clj

```diff
@@ -702,6 +702,16 @@
                        :status http-400-bad-request
                        :log-level :warn})))
 
+    ; validate the backend-proto and health-check-proto combination on same port
+    (let [{:strs [backend-proto health-check-port-index health-check-proto]} service-description-to-use]
+      (when (and backend-proto health-check-port-index health-check-proto
+                 (zero? health-check-port-index) (not= backend-proto health-check-proto))
+        (sling/throw+ {:type :service-description-error
+                       :friendly-error-message (str "The backend-proto (" backend-proto ") and health check proto (" health-check-proto ") "
+                                                    "must match when health-check-port-index is zero")
+                       :status http-400-bad-request
+                       :log-level :warn})))
```

Comment:
This should probably be `:info` instead of `:warn`. We should reserve warnings and errors for when something actually unexpected happens in Waiter, rather than for logging error messages sent back to users with bad requests.

Reply:
Done.